### ISSUE TITLE
Issues due to sitemap generator combined with blog plugin

### ISSUE THREAD

**Issue by Ramonschipper** [ChristopherDosin/Shopware-6-Blog-Plugin#109]
Default shopware command  sitemap:generate giving errors because of a null value passed in to the sitemap handler from the blog plugin.

**Comment by ChristopherDosin** [comment]
@viennt Didn't we solve this issue somewhere?

**Comment by viennt** [comment]
@ChristopherDosin let me check.

**Comment by Ramonschipper** [comment]
@viennt do you have any news for me?

**Comment by Ramonschipper** [comment]
@viennt 
The problem is in the BlogUrlProvider class.
On line 78 to 80 there is a check for setting the loc of a url object.
when this variable is null nog url is set so the sitemap throws an error.
![image](https://user-images.githubusercontent.com/106521028/174022974-80023bac-51e6-4f5e-b290-619038f4abfd.png)

**Comment by viennt** [comment]
@Ramonschipper
Thanks for your time in finding the root cause. I'm working on it now to fix it.

**Comment by viennt** [comment]
Hi @Ramonschipper I can't reproduce this issue, which shopware version are you working on?

**Comment by Ramonschipper** [comment]
Hi @viennt sorry for my late response. 6.4.6.1

**Comment by rklak** [comment]
```
 In Url.php line 63:                                                                                                                                                                                           
   19                                                                                                                                                                                                               
   18   [TypeError]                                                                                                                                                                                                 
   17   Return value of Shopware\Core\Content\Sitemap\Struct\Url::getLoc() must be                                                                                                                                  
   16   of the type string, null returned                                                                                                                                                                           
   15                                                                                                                                                                                                               
   14                                                                                                                                                                                                               
   13 Exception trace:                                                                                                                                                                                              
   12   at /home/stageshop/app-shopware/releases/20221026080635/vendor/shopware/core/Content/Sitemap/Struct/Url.php:63                                                                                              
   11  Shopware\Core\Content\Sitemap\Struct\Url->getLoc() at /home/stageshop/app-shopware/releases/20221026080635/vendor/shopware/core/Content/Sitemap/Service/SitemapExporter.php:168                              
   10  Shopware\Core\Content\Sitemap\Service\SitemapExporter->processSiteMapHandles() at /home/stageshop/app-shopware/releases/20221026080635/vendor/shopware/core/Content/Sitemap/Service/SitemapExporter.php:72   
    9  Shopware\Core\Content\Sitemap\Service\SitemapExporter->generate() at /home/stageshop/app-shopware/releases/20221026080635/vendor/shopware/core/Content/Sitemap/Commands/SitemapGenerateCommand.php:119       
    8  Shopware\Core\Content\Sitemap\Commands\SitemapGenerateCommand->generateSitemap() at /home/stageshop/app-shopware/releases/20221026080635/vendor/shopware/core/Content/Sitemap/Commands/SitemapGenerateCommand
    7  Shopware\Core\Content\Sitemap\Commands\SitemapGenerateCommand->execute() at /home/stageshop/app-shopware/releases/20221026080635/vendor/symfony/console/Command/Command.php:298                              
    6  Symfony\Component\Console\Command\Command->run() at /home/stageshop/app-shopware/releases/20221026080635/vendor/symfony/console/Application.php:1046                                                         
    5  Symfony\Component\Console\Application->doRunCommand() at /home/stageshop/app-shopware/releases/20221026080635/vendor/symfony/framework-bundle/Console/Application.php:96                                     
    4  Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /home/stageshop/app-shopware/releases/20221026080635/vendor/symfony/console/Application.php:299                                        
    3  Symfony\Component\Console\Application->doRun() at /home/stageshop/app-shopware/releases/20221026080635/vendor/symfony/framework-bundle/Console/Application.php:82                                            
    2  Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/stageshop/app-shopware/releases/20221026080635/vendor/symfony/console/Application.php:171                                               
    1  Symfony\Component\Console\Application->run() at /home/stageshop/app-shopware/releases/20221026080635/bin/console:79     
                                                                                                                                                                                                                                                                                                   
    1 sitemap:generate [-i|--salesChannelId [SALESCHANNELID]] [-f|--force]    
```

**Comment by ChristopherDosin** [comment]
@viennt will have a look tomorrow morning.

**Comment by hungmac-sw** [comment]
I will close this issue because we fixed it from v1.5.10
